target/linux/ath79/dts/ar9344_adtran_bsap192x.dtsi

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9344.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	aliases {
+		label-mac-device = &eth0;
+		led-boot = &led_power_amber;
+		led-failsafe = &led_power_amber;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_amber;
+	};
+
+	chosen {
+		/delete-property/ bootargs;
+		bootargs-append = "root=";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins>;
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power_amber: power_amber {
+			label = "amber:power";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_green: power_green {
+			label = "green:power";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wlan_amber {
+			label = "amber:wlan";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		wlan_green {
+			label = "green:wlan";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		lan_amber {
+			label = "amber:lan";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		lan_green {
+			label = "green:lan";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&ref {
+	clock-frequency = <25000000>;
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x0000000 0x0040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x0040000 0x0010000>;
+			};
+
+			partition@50000 {
+				label = "Cert";
+				reg = <0x0050000 0x0040000>;
+				read-only;
+			};
+
+			partition@90000 {
+				label = "Kdump";
+				reg = <0x0090000 0x0170000>;
+				read-only;
+			};
+
+			partition@300000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x0300000 0x1ce0000>;
+			};
+
+			senao: partition@1fe0000 {
+				label = "SENAO";
+				reg = <0x1fe0000 0x0010000>;
+				read-only;
```

Comment:
Add following code here:
```
nvmem-layout {
	compatible = "fixed-layout";
	#address-cells = <1>;
	#size-cells = <1>;

	macaddr_senao_18: macaddr@18 {
		compatible = "mac-base";
		reg = <0x18 0x6>;
		#nvmem-cell-cells = <1>;
	};
};
```
then you can use:
- `nvmem-cells = <&macaddr_senao_18 1>;`
- `nvmem-cells = <&macaddr_senao_18 9>;`
- ...